internal/manifestwork/manifestwork.go

```diff
@@ -175,3 +194,92 @@ func (mwg *manifestWorkGenerator) getOwnedManifestWorks(ctx context.Context, mcm
 
 	return manifestWorkList, err
 }
+
+func (mwg *manifestWorkGenerator) managedClusterKernelMappings(
+	ctx context.Context,
+	mcm hubv1beta1.ManagedClusterModule,
+	kernelVersions []string) []kmmv1beta1.KernelMapping {
+
+	mod := &kmmv1beta1.Module{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mcm.Name,
+			Namespace: mcm.Spec.SpokeNamespace,
+		},
+		Spec: mcm.Spec.ModuleSpec,
+	}
+
+	logger := log.FromContext(ctx)
+
+	mappings := make([]kmmv1beta1.KernelMapping, 0, len(kernelVersions))
+
+	for _, kernelVersion := range kernelVersions {
```

Comment:
Same question as before. Should we iterate over `mcm.Spec.ModuleSpec.ModuleLoader.Container.KernelMappings[]` instead?

Reply:
We are looping over the kernel versions instead of the kernel mappings, because a kernel mapping can be also defined using a regexp. That way one kernel mapping can lead to matching more than one kernel versions. What we would like to have in the resulting `Module` that will be applied on the spoke, is its `KernelMappings` array to include only literal kernel mappings, i.e. one kernel mapping per kernel version. WDYT?

Reply:
That makes sense.